src/pages/student_detail/StudentDetail.tsx

```diff
@@ -36,79 +84,162 @@ const StudentDetail = () => {
+            </InformationContainer>
+            <InformationContainer>
+              <Label>DNI</Label>
+              <StudentInfo>{student.dni}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Fecha de ingreso</Label>
+              <StudentInfo>{student.admissionDate}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Fecha de nacimiento</Label>
+              <StudentInfo>{student.birthday}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Edad</Label>
+              <StudentInfo>{student.age}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Numero de teléfono</Label>
+              <StudentInfo>{student.cellphoneNumber}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Patologías</Label>
+              <StudentInfo>{student.pathologies}</StudentInfo>
+            </InformationContainer>
+          </AllInformationContainer>
+        </StudentInfoContainer>
+        <PaymentInfoContainer>
+          <HeaderBoxes>
+            <SubTitle>
+              <IconStyles>
+                <img src={InfoIcon} alt="info-icon"></img>
+              </IconStyles>
+              Datos de pago y estados
+            </SubTitle>
+            <EditIconStyles>
+              <img src={EditIcon} alt="edit-icon"></img>
+            </EditIconStyles>
+          </HeaderBoxes>
+          <AllInformationContainer>
+            <InformationContainer>
+              <Label>Forma de pago</Label>
+              <StudentInfo>{student.paymentPlan}</StudentInfo>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Estado del alumno</Label>
+              <StudentStatusStyle $status={isActive}>
+                {isActive ? "Activo" : "Inactivo"}
+              </StudentStatusStyle>
+            </InformationContainer>
+            <InformationContainer>
+              <Label>Día de pago</Label>
+              <StudentInfo>
+                {student.paymentDay ||
+                  student.paymentPlan == "Principio de mes"}
+              </StudentInfo>
+            </InformationContainer>
+
+            <InformationContainer>
+              <Label>Situación del alumno</Label>
+              <StudentSituationStyle $situation={student.situation}>
+                {student.situation}
+              </StudentSituationStyle>
+            </InformationContainer>
+            <ButtonContainer>
+              <Button
+                variant="primary"
+                size="large"
+                onClick={() => navigate(MisPlanes)}
+              >
+                Ver cuotas
+              </Button>
+            </ButtonContainer>
+          </AllInformationContainer>
+        </PaymentInfoContainer>
+      </InfoBoxesContainer>
```

Comment:
Todo esto podrías dividirlo en dos funciones, para hacer más legible el código:
```suggestion
      <InfoBoxesContainer>
        <StudentData student={student} navigate={navigate} />
        <PaymentData student={student} navigate={navigate} isActive={isActive} />
      </InfoBoxesContainer>
```

Reply:
La lógica de la info del estudiante y la de los pagos y estados quedaría asi: 
```suggestion
    export default StudentDetail;



const StudentData = ({ student, navigate }: { student: StudentDetailResponse; navigate: (path: string) => void; }) => {
  return (
    <StudentInfoContainer>
      <HeaderBoxes>
        <SubTitle>
          <IconStyles>
            <img
              src={StudentIcon}
              alt="student-icon"
              width={24}
              height={24}
            ></img>
          </IconStyles>
          Datos del alumno
        </SubTitle>
        <EditIconStyles>
          <img
            src={EditIcon}
            alt="edit-icon"
            onClick={() => navigate(EditarAlumno)}
          ></img>
        </EditIconStyles>
      </HeaderBoxes>
      <AllInformationContainer>
        <InformationContainer>
          <Label>Nombre</Label>
          <StudentInfo>{student.name}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Apellido</Label>
          <StudentInfo>{student.lastName}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>DNI</Label>
          <StudentInfo>{student.dni}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Fecha de ingreso</Label>
          <StudentInfo>{student.admissionDate}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Fecha de nacimiento</Label>
          <StudentInfo>{student.birthday}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Edad</Label>
          <StudentInfo>{student.age}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Numero de teléfono</Label>
          <StudentInfo>{student.cellphoneNumber}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Patologías</Label>
          <StudentInfo>{student.pathologies}</StudentInfo>
        </InformationContainer>
      </AllInformationContainer>
    </StudentInfoContainer>
  );
};

const PaymentData = ({ student, navigate, isActive }: { student: StudentDetailResponse; navigate: (path: string) => void; isActive: boolean }) => {
  return (
    <PaymentInfoContainer>
      <HeaderBoxes>
        <SubTitle>
          <IconStyles>
            <img src={InfoIcon} alt="info-icon"></img>
          </IconStyles>
          Datos de pago y estados
        </SubTitle>
        <EditIconStyles>
          <img src={EditIcon} alt="edit-icon"></img>
        </EditIconStyles>
      </HeaderBoxes>
      <AllInformationContainer>
        <InformationContainer>
          <Label>Forma de pago</Label>
          <StudentInfo>{student.paymentPlan}</StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Estado del alumno</Label>
          <StudentStatusStyle $status={isActive}>
            {isActive ? "Activo" : "Inactivo"}
          </StudentStatusStyle>
        </InformationContainer>
        <InformationContainer>
          <Label>Día de pago</Label>
          <StudentInfo>
            {student.paymentDay ||
              student.paymentPlan == "Principio de mes"}
          </StudentInfo>
        </InformationContainer>
        <InformationContainer>
          <Label>Situación del alumno</Label>
          <StudentSituationStyle $situation={student.situation}>
            {student.situation}
          </StudentSituationStyle>
        </InformationContainer>
        <ButtonContainer>
          <Button
            variant="primary"
            size="large"
            onClick={() => navigate(MisPlanes)}
          >
            Ver cuotas
          </Button>
        </ButtonContainer>
      </AllInformationContainer>
    </PaymentInfoContainer>
  )
}
Le pasamos por parámetro la info necesaria y todo lo demás queda igual. Luego llamás a las funciones como te indiqué al principio
```